idaplugin/rematch/network.py

```diff
@@ -34,19 +35,54 @@ def __init__(self, method, url, server=None, token=None, params=None,
     self.token = token
     self.params = params
     self.json = json
+    self.paginate = paginate
+    self.running = True
+    self.started = False
 
     self.signals = WorkerSignals()
 
+  def start(self, callback=None, exception_callback=None):
+    if self.started:
+      raise Exception("query worker already started")
+    self.started = True
+
+    if callback:
+      self.signals.result_dict.connect(callback)
+      self.signals.result_list.connect(callback)
+      self.signals.result_str.connect(callback)
+    if not exception_callback:
+      exception_callback = default_exception_callback
+    self.signals.result_exception.connect(exception_callback)
+    _threadpool.start(self)
+
+  def cancel(self):
+    self.running = False
+
   def run(self):
     try:
-      response = query(self.method, self.url, self.server, self.token,
-                       self.params, self.json)
-      if isinstance(response, dict):
-        self.signals.result_dict.emit(response)
-      elif isinstance(response, list):
-        self.signals.result_list.emit(response)
-      elif isinstance(response, str):
-        self.signals.result_str.emit(response)
+      while self.running:
+        response = query(self.method, self.url, self.server, self.token,
+                         self.params, self.json)
+        if not self.running:
+          break
+
+        if isinstance(response, dict):
+          self.signals.result_dict.emit(response)
+        elif isinstance(response, list):
+          self.signals.result_list.emit(response)
+        elif isinstance(response, str):
+          self.signals.result_str.emit(response)
+
+        # if request is paginated and was successful, automatically request
+        # next page if specified. otherwise, break out of the loop
+        if not self.paginate:
+          break
+
+        if 'next' not in response or not response['next']:
```

Comment:
`self.paginate` should be only true for json responses. For those `response` is a dictionary.

What do you think of a test to make sure `self.json == True` and that response is a dictionary?